### DISCUSSION TITLE
Lovin it

### DISCUSSION THREAD

**Discussion by lowellbert** [cardonabits/haxo-rs#25] [General]
Hi All,

Received my Haxophone a few weeks ago, wasnt expecting it so soon but made for a great Christmas gift to myself. Just had my first time to sit down with it and this thing is great! 

I have been interested in a wind synth for a few years now and was happy to see when the crowd supply orders came up. I am a tenor player and I haven't been able to practice for years as I have a young child who is sleeping when i have free time and life is busy. I wasn't sure how it would feel compared to the real sax, but this feels good and didnt take long to get the basic fingering down. I am still a bit rusty so playing this was also great to not get any out of tune notes to throw my pace off. Great to use my same mouth peice to and I still am blowing the same as I would with my Tenor. This is a great practice unit! love it can't wait to see what I can do with this thing.

A few initial quesitons.

1. Is there a list of the different instrument sounds? I assume on boot it is starting with default Alto or Tenor.
2. I was able to get into the Change sound command, but i noticed on another thread you talked about other commands, I was not able to get those to work, I am using Pi4 
3. Shutdown? i was trying the shutdown command as mentioned but was not working, so I am just pulling the USB cable
4. Back to defaults? After changing my sounds I just wanted to go back to regular sounding Sax, I just rebooted, but wondering if there is a command for that?
5. After about 2 hrs of play time, i noticed i started to get a faint sound with out blowing. Wondering if this is somthing with the Wind sensor IC or just moisture build up or something, not sure if there is any adjusting that or a quick fix?

Thats all, this thing is great!

**Comment by jcard0na** [comment]
Hi @lowellbert ,

Thanks for your kind feedback.  All feedback (good and bad) is welcome and very much appreciated!

Regarding your questions...
> 1. Is there a list of the different instrument sounds? I assume on boot it is starting with default Alto or Tenor.

Check out [this campaign update](https://www.crowdsupply.com/cardona-bits/haxophone/updates/haxophone-sounds-and-colors#haxophone-sounds) for a list of all the pre-installed sounds.  The [default is tenor sax](https://github.com/cardonabits/haxo-rs/blob/main/scripts/systemd/haxo.service#L17).  Also, it is possible to [install a different set of sounds (soundfont file)](https://github.com/cardonabits/haxo-hw/discussions/49) if you have other sounds in mind.

> 2. I was able to get into the Change sound command, but i noticed on another thread you talked about other commands, I was not able to get those to work, I am using Pi4

I try to keep the [instructions and safety guide](https://github.com/cardonabits/haxo-hw/blob/main/docs/instructions_safety.md) up to date (keep an eye on [this issue](https://github.com/cardonabits/haxo-hw/issues/46), there might be a better manual out soon!).  
As you will see, at the moment, only the change instrument command is supported.  The other commands you refer to (I believe shutdown is the only one) were deprecated before the Crowd Supply campaign (this happened after I shut down the haxophone accidentally while playing with friends... :blush: )

> 3. Shutdown? i was trying the shutdown command as mentioned but was not working, so I am just pulling the USB cable

Yep, no shutdown.  This was introduced to prevent SD card corruption if the USB power cable was yanked during a write operation.  Now that the SD card is configured as read-only, there is no risk of corruption, so pulling the USB cable is "the official way" to shut down the haxophone.  If people are interested in bringing that command back, we can explore a better way.  Feel free to open a [poll](https://github.com/cardonabits/haxo-hw/discussions/new?category=polls) if you want to see that implemented.

> 4. Back to defaults? After changing my sounds I just wanted to go back to regular sounding Sax, I just rebooted, but wondering if there is a command for that?

Currently the only way to go back to the previous instrument is to navigate back with `R1/R3` or `R1+R2/R3+R2` keys or (as you discovered already) to reboot.  Maybe it would be a good idea to implement that as well.  Maybe by pressing `R1+R3` or `R1+R2+R3`.  Feel free to open [another poll](https://github.com/cardonabits/haxo-hw/discussions/new?category=polls) for that!

> 5. After about 2 hrs of play time, i noticed i started to get a faint sound with out blowing. Wondering if this is somthing with the Wind sensor IC or just moisture build up or something, not sure if there is any adjusting that or a quick fix?

Is the sound coming from the headphones or from outside (i.e. "the real world")?

After some playing there may be moisture accumulated inside the mouthpiece or on the neck.  I frequently can hear "bubbling sounds" due to that moisture plus air leaking out the mouthpiece/neck interface if there isn't a good seal.  This is where washi tape can help.  But these are "real world" noises, nothing coming from the headphone.

Best,
